src/plugins/irc-events/link.js

```diff
@@ -159,7 +159,14 @@ function fetch(uri, cb) {
 	var limit = Helper.config.prefetchMaxImageSize * 1024;
 	req
 		.on("response", function(res) {
-			if (!(/^image\/.+/.test(res.headers["content-type"]))) {
+			if (/^image\/.+/.test(res.headers["content-type"])) {
+				// response is an image
+				// if Content-Length header reports a size exceeding the prefetch limit, abort fetch
+				const contentLength = parseInt(res.headers["content-length"], 10) || 0;
+				if (contentLength > limit) {
+					res.abort();
```

Comment:
@xPaw, in #1554, we went for `req.abort()`. I wonder if both do the same thing, but should we use the same thing then? Which do you prefer?

Reply:
abort calls abort on req internally, so yes.

Reply:
@dgw, mind switching to `req.abort()` then?

Reply:
There's no `req` parameter passed to this handler, only `res`.

Reply:
You have access to `req` from L160. It's also being used at R179 :)

Reply:
What. Has variable scoping has changed since I learned JS (over a decade ago, I feel old :grin:)? L160 is outside the function definitions in both cases; how does _that_ work? :confused:

(Also, L179 seems silly with the `req.response.req` thing… It's not related to this PR but now I really want to make it just `req.abort()`…)

Reply:
`res.abort` is correct here, don't change it.